ecl/eclcc/eclcc.cpp

```diff
@@ -3090,11 +3097,24 @@ int EclCC::parseCommandLineOptions(int argc, const char* argv[])
         optIgnoreSignatures = true;
     }
 
-    // Append the version to the default main package if the main package has no version
-    if (!optQueryMainPackage.isEmpty() && !optQueryMainPackageVersion.isEmpty())
+    // If no main version is specified it uses the default repo version
+    if (!optDefaultRepo.isEmpty())
+        optQueryMainPackage.set(optDefaultRepo);
+    if (optQueryMainPackageVersion.isEmpty() && !optDefaultRepoVersion.isEmpty())
+        optQueryMainPackageVersion.set(optDefaultRepoVersion);
+
+    if (!optQueryMainPackage.isEmpty())
     {
         if (!strchr(optQueryMainPackage.str(), '#'))
-            optQueryMainPackage.append("#").append(optQueryMainPackageVersion);
+        {
+            if (!optQueryMainPackageVersion.isEmpty())
+                optQueryMainPackage.append("#").append(optQueryMainPackageVersion);
+        }
+    }
+    else
+    {
+        if (!optDefaultRepo.isEmpty())
```

Comment:
Not sure this is this needed given that optQueryMainPackage is being set to optDefaultRepo **in line 3101**?